### DISCUSSION TITLE
warning: cannot find file noahmptable.tb

### DISCUSSION THREAD

**Discussion by mkavulich** [ufs-community/ufs-weather-model#1877] [Q&A]
I am attempting to bring in the latest UFS weather model hash to the UFS SRW application. However, I am getting this error:

```
 Reading in o3data from global_o3prdlos.f77
       oz_coeff =            6
        latsozp =           71
         levozp =           53
         timeoz =           12
An error occurred in ccpp_physics_init
An error occured in GFS_phys_time_vary_init: warning: cannot find file noahmptable.tb

FATAL from PE    80: Call to CCPP physics_init step failed

An error occurred in ccpp_physics_init
An error occured in GFS_phys_time_vary_init: warning: cannot find file noahmptable.tb

FATAL from PE    78: Call to CCPP physics_init step failed

An error occurred in ccpp_physics_init
An error occured in GFS_phys_time_vary_init: warning: cannot find file noahmptable.tb

FATAL from PE    95: Call to CCPP physics_init step failed
```

etc. etc.

I have a few questions about this:

1. Why am I receiving this error even when using the FV3_HRRR CCPP suite, which uses RUC lsm, not noahmp?
2. How should I be referencing this file at runtime? As of #1777, there are multiple copies of `noahmptable.tbl`, one in `ufs-weather-model/tests/parm` and the other in the CCPP submodule (`ufs-weather-model/FV3/ccpp/physics/physics/`). It seems as if documentation was not added as part of this PR. Is this a file I should be linking from one of those locations at runtime? Why is it treated differently from other fix files?

**Comment by mkavulich** [comment]
@barlage I was told you might be able to offer some insight here. They also suggested Cenlin He but I don't know their github handle.

**Comment by barlage** [comment]
@mkavulich I would also suggest @HelinWei-NOAA since it's his PR that added this. First, to address some of your questions, the motivation for doing this is that we are transitioning to using a submodule for noahmp and that submodule will also contain the model parameters, see e.g., https://github.com/NCAR/noahmp/tree/master/parameters; part of this PR #1777  is a step in that direction. 

If you are looking for an example, I suggest looking in global-workflow [here](https://github.com/NOAA-EMC/global-workflow/issues/1614) or [here](https://github.com/NOAA-EMC/global-workflow/pull/1697), specifically the changes to `sorc/link_workflow.sh`. Note the file being accessed is `FV3/ccpp/physics/physics/noahmptable.tbl`.

**Reply by mkavulich** [reply]
@barlage Thanks for the answer. Since this work is still in flux, it seems like a quick and dirty temporary solution (linking the file from `ufs-weather-model/tests/parm` at runtime) is the way to go until the work settles down into its final form. 

This still leaves me with question 1 unanswered: should this table be linked for all runs, even if we are not using NoahMP? The weather model runs seem to be failing if I don't link the table, but this doesn't make sense to me. Is this a bug, or is the table used for processes other than the NoahMP land surface model?

@HelinWei-NOAA Can you offer any insight?

**Reply by barlage** [reply]
@mkavulich, as @HelinWei-NOAA says we will fix this read requirement in an future land PR unless we can squeeze it into someone else's PR earlier. To be clear, this file should not ***need*** to be linked, but in the current code it does need to be. 

I agree with your quick and dirty solution. My only suggestion would be to link `FV3/ccpp/physics/physics/noahmptable.tbl`. They are technically the same file, but really there should be only one file and it should come from the main tracked ccpp physics version. Like I mentioned earlier, the tracked version will eventually be in the noahmp repo so this ccpp version will be the one moved to the repo.

**Comment by HelinWei-NOAA** [comment]
@mkavulich This table should not be linked if we are not using NoahMP. Some initializations of NoahMP are still done in  GFS_phys_time_vary.fv3.F90. We didn't limit open that table only for NoahMP case. We will fix this issue when we have a chance by moving all NoahMP initializations to noahmpdrv.F90.

**Comment by mkavulich** [comment]
Thanks @barlage and @HelinWei-NOAA for your feedback. I will implement a temporary solution in the SRW repo linking the table from the CCPP physics submodule, and will look out for upcoming changes to correct the issues mentioned above.
